### ISSUE TITLE
Module starindexgenerator uses all available cores

### ISSUE THREAD

**Issue by AlbatrosHurleur** [GenomiqueENS/eoulsan#36]
Hi,

I am using Eoulsan 2.5 and noticed using `htop` that in the _starindexgenerator_ module, STAR (2.7.8a) was run with `--runThreadN 56`, which is the maximum available threads on the machine, even if my workflow file is specifying 32:

``` 
    <globals>
        <parameter>
                <name>main.local.threads</name>
                <value>32</value>
        </parameter>
    </globals>
```

The rest of the steps seem to follow the number of threads specified in the XML file.

Please tell me if you need more details about this issue.

Thanks!

**Comment by jourdren** [comment]
Dear @AlbatrosHurleur,

Does this issue only affect the _starindexgenerator_ step or all the others steps (e.g. the mapping step) of your workflow have the same issue?

I think that the following workaround can solve your issue by setting the `requiredprocs` attribute when configuring the _starindexgenerator_ module/step:
```xml
        <!-- Create STAR index -->
        <step requiredprocs="32" skip="false">
                <module>starindexgenerator</module>
                <parameters>
                        <!-- The overhang value must be greater than the length of the reads -->
			<parameter>
				<name>overhang</name>
				<value>100</value>
			</parameter>
			<parameter>
				<name>use.gtf.file</name>
				<value>true</value>
			</parameter>
                        <parameter>
                                <name>gtf.feature.exon</name>
                                <value>exon</value>
                        </parameter>
                        <parameter>
                                <name>gtf.tag.exon.parent.transcript</name>
                                <value>Parent</value>
                        </parameter>
                </parameters>
        </step>
```
Best regards,
Laurent.

**Comment by AlbatrosHurleur** [comment]
Dear @jourdren 

Yes it affects only the _starindexgenerator_ step; the rest of the steps use 32 cores as requested.

Indeed, adding `requiredprocs="32"` to the configuration solved this issue, thank you!
Would it be possible for this step to use the `main.local.threads` parameter in another version to come?

On another note, do you plan to make the newest versions of Eoulsan available from bioconda?

Thanks for your work on this project.

**Comment by jourdren** [comment]
Dear @AlbatrosHurleur,

Great news that the workaround works for you. 
I just fixed the issue with the fc37482f6b60ef59b77cf640b5262c8b35679084 commit.

I will try to package Eoulsan in bioconda as soon as I publish the new Aozan [1,2] major release (3.0). Aozan is my other big project with ToulligQC.

Best regards,

Laurent.

[1] https://www.outils.genomique.biologie.ens.fr/aozan/
[2] https://github.com/GenomiqueENS/aozan/tree/branch-3.0
